### PR TITLE
PROGRAMMES-7053 - main collection page content

### PR DESCRIPTION
**Pull request by Bobkidbob**
https://jira.dev.bbc.co.uk/browse/PROGRAMMES-7053

### CODE REVIEW

**Comment by programmes-developers** [comment]
Can one of the admins verify this patch?

File: src/DsAmen/Presenters/Domain/PromotionCard/promotion_card.html.twig [hunk 1]
```diff
@@ -0,0 +1,17 @@
+<div class="promotion-card box-link" style="background-image:url({{ promotion_card.getImageUrl() }})">
+    <div class="promotion-card__body">
+        <a class="link--block box-link__target cover__meta" href="http://www.bbc.co.uk/news/scotland/scotland_business">
```
Comment (BjornTwachtmann) [inline src/DsAmen/Presenters/Domain/PromotionCard/promotion_card.html.twig]: hardcoded URL

File: src/DsShared/Helpers/TitleLogicHelper.php [hunk 2]
```diff
@@ -57,6 +60,13 @@ public function getOrderedProgrammesForTitle(CoreEntity $programme, ?CoreEntity
             $mainTitleProgramme = array_shift($ancestryArray);
             // Ancestry in TLEO => Series => Episode order is subtitle
             $subTitlesProgrammes = array_reverse($ancestryArray);
+        } elseif ($titleFormat === 'item::ancestry_except_audio_clips') {
+            $mainTitleProgramme = array_shift($ancestryArray);
+            if ($programme instanceof Clip && $programme->getMediaType() === MediaTypeEnum::AUDIO) {
```
Comment (BjornTwachtmann) [inline src/DsShared/Helpers/TitleLogicHelper.php]: there's an isAudio function, might be better

File: templates/find_by_pid/collection.html.twig [hunk 3]
```diff
@@ -1 +1,55 @@
...
+        <ul class="grid">
+            {% for promotion in promotions %}
+                <li class="1/3@gel3b grid__item">
+                    {{ ds_amen('promotionCard', promotion) }}
+                </li>
+            {% endfor %}
+            {% for coreEntity in coreEntities %}
+                <li class="1/2@gel2b 1/3@gel3b grid__item">
+                    {{ ds_amen('coreEntity', coreEntity, {
+                        'title_options': {
```
Comment (BjornTwachtmann) [inline templates/find_by_pid/collection.html.twig]: We'll need a context programme here, coreEntity.getParent() should do it (hopefully)

File: src/Controller/FindByPid/CollectionController.php [hunk 4]
```diff
@@ -13,17 +13,32 @@ class CollectionController extends BaseController
...
-        $relatedLinks = $relatedLinksService->findByRelatedToProgramme($collection, ['related_site', 'miscellaneous']);
+        $page = $this->getPage();
+
+        if ($page === 1) {
+            $promotions = $promotionsService->findActiveNonSuperPromotionsByContext($collection, 4, 1, CacheInterface::MEDIUM);
+        } else {
+            $promotions = [];
+        }
+
+        $coreEntities = $coreEntitiesService->findByGroup($collection, 26, $page);
```
Comment (BjornTwachtmann) [inline src/Controller/FindByPid/CollectionController.php]: V2 uses 24 items, might as well keep that the same

**Pull request merged**